so3/devices/mem.c

```diff
@@ -0,0 +1,41 @@
+#include <vfs.h>
+#include <common.h>
+#include <process.h>
+#include <asm/io.h>
+#include <asm/mmu.h>
+#include <device/driver.h>
+
+#define DEV_CLASS_MEM "mem"
+
+static void *mem_mmap(int fd, addr_t virt_addr, uint32_t page_count, off_t offset)
+{
+	pcb_t *pcb;
+	bool no_cache;
+
+	pcb = current()->pcb;
+
+	/* Use cache only if the target offset is in RAM */
+	no_cache = (offset < mem_info.phys_base) &&
+	           ((mem_info.phys_base + mem_info.size) <= offset);
```

Comment:
```suggestion
	off_t end = offset + page_count * PAGE_SIZE;
        pcb = current()->pcb;
        
	/* Use cache only if the target offset is in RAM */
	no_cache = ((offset < mem_info.phys_base) && 
	           (end < mem_info.phys_base)) || 
	           ((mem_info.phys_base + mem_info.size) <= offset);
```
Would this make sense ?

Reply:
That leads to a preference for activating the cache if any part of the mapped range is in the RAM. I personnaly think it would make more sense to disable it if any part of the mapped range is out of the RAM as having cache on devices can lead to *undefined behavior*. E.g. when a register is updated by the device itself, it doesn't reflect in the cache.

In any case, this is only for cases where the mapped range is on the RAM edges, which I don't see why it would be done. That's why I was fine with my first proposal (which I just found out that I forgot to invert the `&&` to `||` 🤦🏽‍♂️).

I will make a more robust version where the mapping is break down to activate the cache only for the RAM even if the both are target by the range